module/zfs/zfs_ioctl.c

```diff
@@ -5821,6 +5822,44 @@ zfs_ioc_send_space(const char *snapname, nvlist_t *innvl, nvlist_t *outnvl)
 	return (error);
 }
 
+/*
+ * Sync the currently open TXG to disk for the specified pool.
+ * This is somewhat simular to 'zfs_sync()'.
+ * For all cases that do not result in error this iotcl will wait for
+ * the currently open TXG to commit before returning back to the caller.
+ *
+ * innvl is unused
+ *
+ * onvl is unused
+ */
+/* ARGSUSED */
+static int
+zfs_ioc_pool_sync(const char *pool, nvlist_t *innvl, nvlist_t *onvl)
+{
+	int err;
+	spa_t *spa;
+
+	ASSERT3P(innvl, ==, NULL);
+
+	/* we don't want a panicking kernel writing to storage */
+	if (unlikely(oops_in_progress))
+		return (EHOSTDOWN);
+
+	if ((err = spa_open(pool, &spa, FTAG)) != 0)
+		return (err);
+
+	/* we checked these as part of the ioctl call */
+	ASSERT3S(spa_writeable(spa), ==, B_TRUE);
+	ASSERT3S(spa_suspended(spa), ==, B_FALSE);
+	ASSERT3S(spa_state(spa), ==, POOL_STATE_ACTIVE);
```

Comment:
I'm not a fan of this. It's incredibly unlikely but a pool could become suspended between the ioctl handler's check and the assertion's check. No locks are held in the interim between these checks.